apis/v1alpha1/gateway_types.go

```diff
@@ -435,7 +435,7 @@ type RouteBindingSelector struct {
 	//
 	// Support: Core
 	//
-	// +kubebuilder:default={onlySameNamespace:true}
+	// +kubebuilder:default={from:SameNamespace}
```

Comment:
Should we remove this default here and instead rely on controllers to interpret the default?

Reply:
I can see a use cases for allowing all ns's or same ns only as a default, so +1 on not setting a default.

Reply:
I think it could make sense to remove the default but I think we'd still want to define a consistent default behavior between controllers (likely still the `from:SameNamespace` equivalent).

Reply:
Updated to remove the default now.